### ISSUE TITLE
SVG map not replaced

### ISSUE THREAD

**Issue by almetge** [ace-design/island#59]
After each run, if a file "map.svg" exists, the engine does not replace it with the new one. I have to delete this file by hand in order to be able to get the new one.
Note that the file is still generated in the Windows temp folder.
I'm using Java 1.8 under Windows 10.

**Comment by mosser** [comment]
Ok, sounds like an ongoing work when migrating from java.io to java.nio. Axctually File.rename might not move a file when several partition are involved. But not always, making it hard to reproduce.

**Comment by mosser** [comment]
Identified points: 

azrael:island mosser$ find . ( -name '_.scala' -o -name '_.java' ) | xargs grep rename
./engine/src/main/scala/eu/ace_design/island/dsl/DiSLand.scala:      result._2.renameTo(new java.io.File(result._1))
./engine/src/main/scala/Main.scala:    viewer(m).renameTo(new java.io.File(name+".svg"))
./engine/src/test/scala/eu/ace_design/island/map/processes/ProcessTestTrait.scala:    pdf(result).renameTo(new File(name))
./runner/src/main/java/eu/ace_design/island/runner/Runner.java:     viewer.apply(theIsland).renameTo(new File(outputDir.getPath() + "/_pois.json"));

**Comment by mosser** [comment]
I am unable to reproduce the bug :(.

I wrote a test to ensure that after two run, the map.svg file is updated in the output directory.
[https://github.com/ace-design/island/blob/develop/runner/src/test/java/RunnerTest.java](https://github.com/ace-design/island/blob/develop/runner/src/test/java/RunnerTest.java)

Could you please check that the test is red when executed in your environment ?

**Comment by almetge** [comment]
Your test pass, both on Windows and Ubuntu.
Also, your last commit fixed the bug, at least for Windows.

When using it on my Ubuntu, I get a FileSystemException, but I believe this happen because I am using a different filesystem to store my files (a shared partition between Windows and Linux, /data is the mountpoint):

```
java.nio.file.FileSystemException: /data/Dropbox/Polytech/SI3/QGL/./map.svg: Opération non permise
java.nio.file.FileSystemException: /data/Dropbox/Polytech/SI3/QGL/./map.svg: Opération non permise
    at sun.nio.fs.UnixException.translateToIOException(UnixException.java:91)
    at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:102)
    at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:107)
    at sun.nio.fs.UnixCopyFile.copyFile(UnixCopyFile.java:283)
    at sun.nio.fs.UnixCopyFile.move(UnixCopyFile.java:471)
    at sun.nio.fs.UnixFileSystemProvider.move(UnixFileSystemProvider.java:262)
    at java.nio.file.Files.move(Files.java:1395)
    at eu.ace_design.island.runner.Runner.exportMap(Runner.java:292)
    at eu.ace_design.island.runner.Runner.processResults(Runner.java:231)
    at eu.ace_design.island.runner.Runner.startEngine(Runner.java:205)
    at eu.ace_design.island.runner.Runner.fire(Runner.java:150)
    at fr.unice.polytech.qgl.qba.Main.main(Main.java:23)
```

I changed your test so it doesn't use a temporary folder, and it failed with the same exception.

When I created a map.svg by hand, after starting the runner (and having this exception), the file disapeared, so maybe you could catch this exception in your runner, and use renameTo, which worked when no file was present? Or just delete the file, and keep using renameTo.

**Comment by mosser** [comment]
RenameTo is not safe as is might not move the file at all => cannot use it.

The CI server is based on a linux distro, and the test works fine on it => it looks like the issue depends of your very specific environment. Which is also extremely weird is that it works for the other file (_pois.json) that is copied using the very same method. 

You should put a breakpoint in the Runner at line 292 and clearly identify which IOException is thrown by the code on your system.

**Comment by almetge** [comment]
I never noticed the _pois.json file since I never used it.

I'm sorry if I didn't made it clear, but your test does pass on my Ubuntu.

The above error occur when trying to run our explorer. I couldn't get any more information than the stacktrace.
I tried to change the Main I'm using to run our explorer: I used `.storingInto("/home/alexis/test/")` twice, and the files were copied and replaced without any problem. But when reverting to my original configuration, with `.storingInto("/data/Dropbox/Polytech/SI3/QGL")`, I get the FileSystemException on the first run.

So, it definitely is a bug on my side, which could be caused by the usage of two filesystem: 
- The root filesystem, in which is /tmp/ and /home/alexis/, is an ext4 filesystem
- The /data filesystem, which is a NTFS filesystem

This is odd, but it seems that `Files.move()` is just like `renameTo()` when using two differents filesystems.
